### PR TITLE
[Tools] MRI Protocol Rerunner

### PR DESCRIPTION
**Pull request by davidblader**
This new script Reruns MRI protocol checks on all existing imaging files (MNC) inserted into LORIS.  
Protocol violations are output to a new file: `newly_violated_scans_$date.txt` 

It selects the contents of the `files` table and checks each registered file against the MRI protocol parameters. The scan parameters are checked using the `getAcquisitionProtocol()` function, with `bypass_extra_checks` set to false(0). 

If a given scan does not match any protocol entry (scan type `unknown`), this "failed" scan is logged in `newly_violated_scans_$date.txt`  (date is just %Y -%m -%d)

How to test: 
- [ ] Run the script, then verify that violated scans are properly flagged.  
- [ ]   

To do: 
- [ ] Update/clean documentation in script (much copied from `imaging_upload_file.pl`)
- [ ] Protocol group interaction? (#369)

### CODE REVIEW

**Comment by christinerogers** [comment]
@cmadjar I took a stab at the PR header, including To Do (a little cleaning)

File: tools/MRIProtocolRerunner.pl [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+   4) Remove the uploaded file once the previous steps have completed
+   5) Update the mri_upload table
+
+Documentation: perldoc imaging_upload_file.pl
+
+HELP
+my $Usage = <<USAGE;
+usage: $0 </path/to/UploadedFile> -upload_id [options]
+       $0 -help to list options
+Note:  Please make sure that the </path/to/UploadedFile> and the upload_id
```
Comment (PapillonMcGill) [inline tools/MRIProtocolRerunner.pl]:
```suggestion
Note:  Please ensure that the </path/to/UploadedFile> and the upload_id
```
more formal

File: tools/MRIProtocolRerunner.pl [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+my $TmpDir = tempdir(
+    $template, TMPDIR => 1, CLEANUP => 1
+);
+
+my @temp     = split(/\//, $TmpDir);
+my $templog  = $temp[$#temp];
+my $LogDir   = "$data_dir/logs";
+if (!-d $LogDir) {
+    mkdir($LogDir, 0770);
+}
```
Comment (PapillonMcGill) [inline tools/MRIProtocolRerunner.pl]: What if the directory exist but the permissions are not properly set?
Should also check if the script can write to that directory

**Comment by cmadjar** [comment]
Closing for now. Someone will pick up that work on a separate branch

**Pull request closed without merging**